stat.go

```diff
@@ -648,6 +648,71 @@ func KullbackLeibler(p, q []float64) float64 {
 	return kl
 }
 
+// OrdinaryLinearRegression computes the simple linear regression of x and y.
+// It returns the values for \hat{\alpha}, \hat{\beta} and the coefficient
+// of determination, r. If origin is true, the regression is forced to pass
+// through the origin.
+// The lengths of x and y must be equal. If weights is nil then all of the
+// weights are 1. If weights is not nil, then len(x) must equal len(weights).
```

Comment:
It seems better to not have r^2 returned here and just have the separate function
````
RSquared(x, y []float64, a, b float64) float64
````

Reasons:
a) There should be a way to calculate R^2 from a general line, as there are many ways to get a line.
b) Calculating R here doesn't save calculation time as opposed to calculating it in a separate function
c) Given a and b, the functionalities seem orthogonal, and separate functions show that better
d) R^2 isn't the only measure of goodness of fit (just the standard one for lines), and including it here gives R^2 a special significance